### ISSUE TITLE
Agent Tanuki Wrapper license keys on master do not appear to be working

### ISSUE THREAD

**Issue by chadlwilson** [gocd/gocd#9547]
##### Issue Type

- Bug Report

##### Summary

Since the commit from @arvindsv in #9403 and https://github.com/gocd/gocd/commit/bb410c8a21b18f65a9e768b1683803cdbfa0478b the agent wrapper licenses appear to not working on `master`, or at least `wrapper-license-relative-path-go-agent.conf` which is used within the agent Docker builds. I'm not sure about `wrapper-license-linux-go-agent.conf`. `wrapper-license-relative-path-go-server.conf` for the server seems to be OK, by contrast.

Reverting the changes to the license to the previous, nominally expired license, seems to allow the container to start, but I am not sure what that means or why that would be.

This is causing all docker agent builds to fail as part of `installers` pipeline, both locally and on the pipeline.

##### Steps to Reproduce

1. `./gradlew docker:gocd-server:alpine-3.11:assemble -PdockerBuildLocalZip` will fail
2. To see the container output and reason for failure prior to #9546 being merged, run `docker run --rm -e GO_SERVER_URL=http://localhost:8153/go --name gocd-agent-alpine-3.11 gocd-agent-alpine-3.11:v21.3.0-12572` to run the image build slightly earlier (may need to adjust the tag based on # commits)

##### Expected Results
Should be able to start agent containers built off `master` and get a passing Gradle build.

##### Actual Results

Container fails to start, causing the tests to fail, and the docker image build to fail.

##### Possible Fix
- Temporarily revert the license key changes for agents made in https://github.com/gocd/gocd/commit/bb410c8a21b18f65a9e768b1683803cdbfa0478b ?
- I tried swapping them around in case there was a copy+paste error, but this didn't seem to change anything.

##### Log snippets

When starting the agent containers:
```
/docker-entrypoint.sh: Creating directories and symlinks to hold GoCD configuration, data, and logs
$ mkdir -v /godata/config
created directory: '/godata/config'
$ ln -sv /godata/config /go/config
'/go/config' -> '/godata/config'
$ mkdir -v /godata/logs
created directory: '/godata/logs'
$ ln -sv /godata/logs /go/logs
'/go/logs' -> '/godata/logs'
$ mkdir -v /godata/pipelines
created directory: '/godata/pipelines'
$ ln -sv /godata/pipelines /go/pipelines
'/go/pipelines' -> '/godata/pipelines'
/docker-entrypoint.sh: Creating directories and symlinks to hold GoCD wrapper binaries
$ ln -sv /go-agent/bin /go/bin
'/go/bin' -> '/go-agent/bin'
$ ln -sv /go-agent/lib /go/lib
'/go/lib' -> '/go-agent/lib'
$ ln -sv /go-agent/run /go/run
'/go/run' -> '/go-agent/run'
$ ln -sv /go-agent/wrapper /go/wrapper
'/go/wrapper' -> '/go-agent/wrapper'
$ ln -sv /go-agent/wrapper-config /go/wrapper-config
'/go/wrapper-config' -> '/go-agent/wrapper-config'
$ cp -rfv /go-agent/config/agent-bootstrapper-logback-include.xml /go/config/agent-bootstrapper-logback-include.xml
'/go-agent/config/agent-bootstrapper-logback-include.xml' -> '/go/config/agent-bootstrapper-logback-include.xml'
$ cp -rfv /go-agent/config/agent-launcher-logback-include.xml /go/config/agent-launcher-logback-include.xml
'/go-agent/config/agent-launcher-logback-include.xml' -> '/go/config/agent-launcher-logback-include.xml'
$ cp -rfv /go-agent/config/agent-logback-include.xml /go/config/agent-logback-include.xml
'/go-agent/config/agent-logback-include.xml' -> '/go/config/agent-logback-include.xml'
/docker-entrypoint.sh: Running custom scripts in /docker-entrypoint.d/ ...
$ sed -i -e s@wrapper.logfile=.*@/wrapper.logfile=/go/logs/go-agent-bootstrapper-wrapper.log@g -e s@wrapper.java.command=.*@wrapper.java.command=/gocd-jre/bin/java@g -e s@wrapper.working.dir=.*@wrapper.working.dir=/go@g /go-agent/wrapper-config/wrapper.conf
$ exec /usr/local/sbin/tini -g -- /go/bin/go-agent console
Running go-agent...
wrapper  | License Error:
wrapper  | The License Key, 202105080000006, was found but failed to be validated.
wrapper  | Please verify that the wrapper.license.* properties match exactly and have
wrapper  | not been modified.  Also verify that the following property values match
wrapper  | those specified when generating the license:
wrapper  |   wrapper.java.mainclass=org.tanukisoftware.wrapper.WrapperJarApp
wrapper  |   wrapper.app.parameter.1=lib/agent-bootstrapper.jar
wrapper  |
wrapper  | The Java Service Wrapper requires a License Key to activate the
wrapper  | software.  Licenses can be purchased on the Java Service Wrapper
wrapper  | web site:
wrapper  |   http://wrapper.tanukisoftware.com/purchase
wrapper  |
wrapper  | You can also immediately obtain a one-month Free trial license:
wrapper  |   http://wrapper.tanukisoftware.com/trial
wrapper  |
wrapper  | License Keys can be generated for the following HostId:
wrapper  |   HostId #1 (eth0):       0242ac110002
wrapper  |
wrapper  | You may include license keys for several servers within a single file by
wrapper  |   prefixing the property names of each key with their respective
wrapper  |   server host name or HostId. Please use the following syntax:
wrapper  |   wrapper.<hostname|hostid>.license.*.
wrapper  |   The host name of this machine is: 1fadfa1aaa38
wrapper  |
wrapper  | --> Wrapper Started as Console
wrapper  | Java Service Wrapper Standard Edition 64-bit 3.5.41
wrapper  |   Copyright (C) 1999-2019 Tanuki Software, Ltd. All Rights Reserved.
wrapper  |     http://wrapper.tanukisoftware.com
wrapper  |
wrapper  | <-- Wrapper Stopped
```

**Comment by arvindsv** [comment]
Thanks, @chadlwilson. I'll look at this in my evening. I did copy the new license from their site, but maybe it's wrong.

**Comment by chadlwilson** [comment]
Can now be seen with log output [here](https://build.gocd.org/go/tab/build/detail/installers/2568/docker/2/docker-agent).

**Comment by arvindsv** [comment]
I've reproduced this on a normal agent installer, by replacing the working license with the new (wrong?) one. I've written to the license management group, for me to get access to the Tanuki site, so that I can follow up on this. If everything looks ok from that angle, I'll work with Tanuki support on this. Will keep you updated. Thank you.

**Comment by chadlwilson** [comment]
Hi @arvindsv - what do you think about temporarily reverting the agent license changes on master until we can get to the bottom of this? The old _expired_ license seems to actually work (not sure why this might be - if this is an illusion, let me know).

I'm a bit nervous about merging any other PRs since this is blocking the pipeline at the `installers` group which means the smoke tests and regression tests don't run; so not great practice to be pushing on top of.

Currently we have [this set of changes](https://build.gocd.org/go/compare/installers/2561/with/2575) which are err, un-smoked on master.

**Comment by arvindsv** [comment]
Makes sense. Go ahead. I think we will be breaching the license if we
release. But, no harm in temporarily reverting it. I’m away today, but can
help tomorrow if needed. Thank you!

**Comment by arvindsv** [comment]
@chadlwilson I've found what I think is the correct license key. I've pasted it below:

```
#encoding=UTF-8
wrapper.license.type=DEV
wrapper.license.id=202105080000005
wrapper.license.licensee=ThoughtWorks
wrapper.license.group=Go
wrapper.license.dev_application=GoCD Agent
wrapper.license.features=64bit
wrapper.license.upgrade_term.begin_date=2008-06-03
wrapper.license.upgrade_term.end_date=2022-06-03
wrapper.license.key.1=d654-fc5a-80c1-b244
wrapper.license.key.2=973e-f24c-3ab3-50ba
wrapper.license.key.3=81e1-7250-a620-449e
wrapper.license.key.4=c6dd-9cee-ce51-1d14
```

The older one (202105080000006) had an incorrect value it was expecting for `wrapper.app.parameter.1`. It was expecting the value of that parameter to be `../lib/agent-bootstrapper.jar`. The correct value should be `lib/agent-bootstrapper.jar` (as seen in `wrapper.conf`). That's the value this license key expects.

My laptop is away for repairs till tomorrow. I'll create a PR for it on Friday or whenever it comes back. If you want to do it / try it out earlier, please feel free to do that. I'm just planning to recreate https://github.com/gocd/gocd/pull/9403 with these values in the file `installers/include/wrapper-license-relative-path-go-agent.conf`.

**Comment by chadlwilson** [comment]
Sure thing, I'll do that. So you think the non-relative/Linux one is probably fine? Not sure 9f the installer tests verify that, but hopefully I'll find out :)

**Comment by arvindsv** [comment]
> So you think the non-relative/Linux one is probably fine?

I think so. The installer-tests pipeline (https://build.gocd.org/go/pipelines/installer-tests/1365/install-tests/1) also does a bunch of verification. Not sure if it will catch this ... but let's see. Once I get my laptop back, I can verify all the installers manually just to be very sure.

There's also a bundled-JRE related issue I need to look at anyway.

**Comment by chadlwilson** [comment]
Yeah, on a related note, I think we probably need to kill off Alpine `3.11` agent images (EOL in Nov 2021), add `3.14` (June 2021) and possibly change the server default base distro to `3.14` as well (server default base image is currently on Alpine `3.11` which is not great given it's going EOL soon).

**Comment by arvindsv** [comment]
Tried to create a "discussion" for this: https://github.com/gocd/gocd/discussions/9576
